### PR TITLE
Upgrade to Intern 4 and Dojo 2 beta 2.4

### PR DESCRIPTION
**Pull request by bryanforbes**
**Type:** enhancement

The following has been addressed in the PR:

* [ ] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

**Description:**

Upgrades `@dojo/streams` to Intern 4 and Dojo2 beta 2.4. I have left the `tests: {}` blocks unindented to make it easier to review the changes and will indent them after approval.

### CODE REVIEW

File: src/ReadableStreamReader.ts [hunk 1]
```diff
@@ -129,7 +129,7 @@ export default class ReadableStreamReader<T> {
 				stream.pull();
 			}
 			return Promise.resolve({
-				value: chunk,
+				value: <any> chunk,
```
Comment (rorticus) [inline src/ReadableStreamReader.ts:132]: Can this not be a `T` ?

File: .travis.yml [hunk 2]
```diff
@@ -17,9 +17,7 @@ install:
 - travis_retry npm install
 script:
 - grunt
-- grunt intern:node --combined
-- grunt intern:saucelabs --combined
-- grunt remapIstanbul:ci
+- grunt intern:browserstack --test-reporter
```
Comment (rorticus) [inline .travis.yml:20]: We're switching from saucelabs to browserstack here. Is that intentional?

File: tests/unit/ReadableStream.ts [hunk 3]
```diff
@@ -155,6 +154,7 @@ registerSuite({
 			};
 		},
 
+		tests: {
```
Comment (rorticus) [inline tests/unit/ReadableStream.ts]: Do the lines in these `{}` need to be indented?

File: tests/unit/ReadableStreamReader.ts [hunk 4]
```diff
@@ -129,7 +129,7 @@ export default class ReadableStreamReader<T> {
 				stream.pull();
 			}
 			return Promise.resolve({
-				value: chunk,
+				value: <any> chunk,
```
Comment (rorticus) [inline tests/unit/ReadableStreamReader.ts]: Can this not be a `T` ?

**Review by rorticus — CHANGES_REQUESTED** [review]
This looks great! I did have one style request though. It looks like most of the tests in which you added the `tests` key have the wrong indentation, i.e., the tests in side `tests` are not indented.

File: .travis.yml (hunk 2 above)
Comment (bryanforbes) [reply .travis.yml:20]: Yes

File: tests/unit/ReadableStream.ts (hunk 3 above)
Comment (bryanforbes) [reply tests/unit/ReadableStream.ts]: Yes, this was simply for review purposes. I'll indent them before merging.

File: tests/unit/ReadableStreamReader.ts (hunk 4 above)
Comment (bryanforbes) [reply tests/unit/ReadableStreamReader.ts]: Yes, I'll update.

**Comment by bryanforbes** [comment]
I've updated the PR based on your feedback. This PR will also address dojo/meta#211.

File: src/ReadableStreamReader.ts (hunk 1 above)
Comment (bryanforbes) [reply src/ReadableStreamReader.ts:132]: Yes, it can. I've updated in in my latest push.

**Review by rorticus — APPROVED** [review]

**Pull request merged**